src/client.js

```diff
@@ -136,14 +137,27 @@ export default class StitchClient {
    */
   authenticate(providerType, options = {}) {
     // reuse existing auth if present
-    return this.auth.getAccessToken().then(accessToken => {
+    return Promise.all(
+      [
+        this.auth.getAccessToken(),
+        this.auth.getLoggedInProviderType()
+      ]
+    ).then(([accessToken, loggedInProviderType]) => {
       if (accessToken) {
-        return this.auth.authedId();
+        if (providerType === PROVIDER_TYPE_ANON && loggedInProviderType === PROVIDER_TYPE_ANON) {
+          return false; // is authenticated, skip log in
+        }
+
+        return this.logout(); // will not be authenticated, continue log in
       }
 
-      return this.auth.provider(providerType).authenticate(options)
-        .then(() => this.auth.authedId());
-    });
+      return true; // is not authenticated, continue log in
+    }).then((isNotAuthed) => {
+      if (isNotAuthed) {
```

Comment:
[nit] Feels like the name of `isNotAuthed` is more apt as `shouldAuth`.

Reply:
Fair point.